tests/components/iskra/test_config_flow.py

```diff
@@ -0,0 +1,241 @@
+    assert result["title"] == f"{SG_MODEL} {SERIAL}"
+    assert result["data"][CONF_HOST] == HOST
+
+
+async def test_user_rest_auth(hass: HomeAssistant, mock_pyiskra_rest) -> None:
+    """Test the user flow with Rest API protocol and authentication required."""
+    mock_pyiskra_rest.side_effect = NotAuthorised
+
+    # Test if prompted to enter username and password if not authorised
+    result = await hass.config_entries.flow.async_init(
+        "iskra",
+        context={"source": SOURCE_USER},
+        data={CONF_HOST: HOST, CONF_PROTOCOL: "rest_api"},
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "authentication"
+
+    # Test failed authentication
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_USERNAME: "", CONF_PASSWORD: ""},
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {"base": "invalid_auth"}
+    assert result["step_id"] == "authentication"
+
+    # Test successful authentication
+    mock_pyiskra_rest.side_effect = None
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_USERNAME: "", CONF_PASSWORD: ""},
+    )
+
+    # Test successful Rest API configuration
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == f"{SG_MODEL} {SERIAL}"
+    assert result["data"][CONF_HOST] == HOST
+
+
+async def test_user_modbus(hass: HomeAssistant, mock_pyiskra_modbus) -> None:
+    """Test the user flow with Modbus TCP protocol."""
+
+    result = await hass.config_entries.flow.async_init(
+        "iskra",
+        context={"source": SOURCE_USER},
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_init(
+        "iskra",
+        context={"source": SOURCE_USER},
+        data={CONF_HOST: HOST, CONF_PROTOCOL: "modbus_tcp"},
+    )
+
+    # Test if propmpted to enter port and address
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "modbus_tcp"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_PORT: MODBUS_PORT,
+            CONF_ADDRESS: MODBUS_ADDRESS,
+        },
+    )
+
+    # Test successful Modbus TCP configuration
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == f"{PQ_MODEL} {SERIAL}"
+    assert result["data"][CONF_HOST] == HOST
+
+
+async def test_abort_if_already_setup(hass: HomeAssistant, mock_pyiskra_rest) -> None:
+    """Test we abort if Iskra is already setup."""
+
+    MockConfigEntry(domain="iskra", unique_id=SERIAL).add_to_hass(hass)
+    result = await hass.config_entries.flow.async_init(
+        "iskra",
+        context={"source": SOURCE_USER},
+        data={CONF_HOST: HOST, CONF_PROTOCOL: "rest_api"},
+    )
```

Comment:
should we parametrize this check to also check if this is checked when we go for modbus?

Reply:
parametrized